spike-worker/bulkImport.js

```diff
@@ -35,14 +37,30 @@ const doBulkInsert = (csvPayload) => {
 
   // setup your columns - always indicate whether the column is nullable
   // TODO add all fields that are being imported to this list
-  bulkLoad.addColumn('firstName', TYPES.Int, { nullable: false })
-  bulkLoad.addColumn('lastName', TYPES.NVarChar, { length: 50, nullable: true })
-  // etc....
+  bulkLoad.addColumn('school_id', TYPES.Int, { nullable: false })
+  bulkLoad.addColumn('upn', TYPES.Char, { length: 13, nullable: false })
+  bulkLoad.addColumn('lastName', TYPES.NVarChar, { nullable: false })
+  bulkLoad.addColumn('foreName', TYPES.NVarChar, { nullable: false })
+  bulkLoad.addColumn('middleNames', TYPES.NVarChar, { nullable: true })
+  bulkLoad.addColumn('gender', TYPES.Char, { length: 1, nullable: false })
+  bulkLoad.addColumn('dateOfBirth', TYPES.DateTimeOffset, { nullable: false })
+
+  // Fetch all school ids for pupil records
+  const schoolDfeNumbers = csvPayload.map(r => `${r[0]}${r[1]}`)
+  const schools = await schoolDataService.sqlFindByDfeNumbers(schoolDfeNumbers)
 
   for (let index = 0; index < csvPayload.length; index++) {
     const csvRow = csvPayload[index]
     // TODO map all fields into the object
-    bulkLoad.addRow({ firstName: csvRow.firstName, lastName: csvRow.lastName })
+    bulkLoad.addRow({
+      school_id: schools.find(s => s.dfeNumber === parseInt(`${csvRow[0]}${csvRow[1]}`)),
```

Comment:
would it not be best to run through this first and establish if any of the schools in the csv file do not exist in the database?

Reply:
@GuyHarwood indeed, I have updated that part.